src/Text/Parsing/StringParser/CodeUnits.purs

```diff
@@ -62,10 +74,16 @@ anyDigit = try do
 
 -- | Match the specified string.
 string :: String -> Parser String
-string nt = Parser \s ->
-  case s of
-    { str, pos } | SCU.indexOf' (Pattern nt) pos str == Just pos -> Right { result: nt, suffix: { str, pos: pos + SCU.length nt } }
-    { pos } -> Left { pos, error: ParseError ("Expected '" <> nt <> "'.") }
+string nt =
+  let
+    ntLength = SCU.length nt
+  in
+    Parser \{ str, pos } ->
+      case nt == (SCU.take ntLength (SCU.drop pos str)) of
```

Comment:
I think using an if-then-else expression is more idiomatic than using a `case` statement where you match on `true` and `false`. But beyond that, I think it would be best to revert this function to its old implementation, since (as far as I can tell) the old implementation works, and it's best to keep changes minimal. I also think `indexOf` is preferable to `drop` followed by `take` here, since it accomplishes what we need more directly.